src/Api.js

```
@@ -32,7 +32,7 @@ class Api {
     this.token = null;
     this.apiReference = apiServices.register(this);
     this.baseUrl = (() => {
-      if (baseUrl.substr(-1) !== "/") {
+      if (typeof baseUrl === "string" && baseUrl.substr(-1) !== "/") {
```

I'd rather avoid checking types this way.
Do we requite `baseUrl` here? If not just check if `baseUrl` has a value, otherwise let it crash for non-string values or throw an error with a friendly message.